account-kit/infra/src/middleware/gasManager.ts

```diff
@@ -172,22 +172,36 @@ export function alchemyGasAndPaymasterAndDataMiddleware(
         ),
         ...(account.getEntryPoint().version === "0.7.0"
           ? {
-              paymasterVerificationGasLimit: overrideField<"0.7.0">(
-                "paymasterVerificationGasLimit",
-                overrides_ as UserOperationOverrides<"0.7.0">,
-                feeOptions,
-                userOp
-              ),
-              paymasterPostOpGasLimit: overrideField<"0.7.0">(
-                "paymasterPostOpGasLimit",
-                overrides_ as UserOperationOverrides<"0.7.0">,
-                feeOptions,
-                userOp
-              ),
-            }
+            paymasterVerificationGasLimit: overrideField<"0.7.0">(
+              "paymasterVerificationGasLimit",
+              overrides_ as UserOperationOverrides<"0.7.0">,
+              feeOptions,
+              userOp
+            ),
+            paymasterPostOpGasLimit: overrideField<"0.7.0">(
+              "paymasterPostOpGasLimit",
+              overrides_ as UserOperationOverrides<"0.7.0">,
+              feeOptions,
+              userOp
+            ),
+          }
           : {}),
       });
 
+      if (isHex(overrides.maxFeePerGas)) {
+        userOp.maxFeePerGas = overrides.maxFeePerGas;
+      }
+      if (isHex(overrides.maxPriorityFeePerGas)) {
+        userOp.maxPriorityFeePerGas = overrides.maxPriorityFeePerGas;
+      }
+      if (isHex(overrides.verificationGasLimit)) {
+        userOp.verificationGasLimit = overrides.verificationGasLimit;
+      }
+      if (isHex(overrides.callGasLimit)) {
+        userOp.callGasLimit = overrides.callGasLimit;
+      }
```

Comment:
I think we have an `applyOverride` helper function so you could do something like:

```ts
userOp = {
...userOp
field: applyOverride(overrides, userOp) // or w/e the api is for that helper
}